drivers/ethernet/eth_stm32_hal.c

```diff
@@ -1187,6 +1189,103 @@ static int eth_initialize(const struct device *dev)
 	return 0;
 }
 
+static uint32_t reverse(uint32_t val)
```

Comment:
Can it be replaced by `sys_be32_to_cpu` ?

Reply:
Hi @erwango thanks for your comment. Unfortunately I don't think this is possible because the `sys_be32_to_cpu` is a byte reversal operation but on the other hand what is required is a bitwise reversal as per the STM32H7 series reference manual. However, one thing I'm not a 100% sure is this bit reversal only for STM32H7? For STM32F4 series it wasn't mentioned in the reference manual. If you could possibly confirm this that would be greatly appreciated!

![image](https://user-images.githubusercontent.com/16928715/212885385-b6a677db-8cc4-42b6-b045-07169da91d7a.png)

Reply:
@cperera-aud I'm checking this internally

Reply:
Thanks @erwango!

Reply:
@cperera-aud It is confirmed to be H7 specific.